src/access.lua

```diff
@@ -94,30 +99,30 @@ function _M.execute(conf)
   local body, err = sock:receive(tonumber(headers['content-length']))
   if err then
     ngx.log(ngx.ERR, name .. "failed to read body " .. host .. ":" .. tostring(port) .. ": ", err)
-    return
+    return kong_response.exit(500, "failed to read body")
   end
 
   ok, err = sock:setkeepalive(conf.keepalive)
   if not ok then
     ngx.log(ngx.ERR, name .. "failed to keepalive to " .. host .. ":" .. tostring(port) .. ": ", err)
-    return
+    return kong_response.exit(500, "failed to keepalive to host")
   end
 
-  if status_code > 299 then
-    if err then 
-      ngx.log(ngx.ERR, name .. "failed to read response from " .. host .. ":" .. tostring(port) .. ": ", err)
+  if status_code == 200 then
```

Comment:
it can also be 204.... I think you should check 200 <= status_code <= 399

Reply:
Remove condition

Reply:
Done